internal/pkg/agent/application/upgrade/step_mark.go

```diff
@@ -335,8 +375,11 @@ func saveMarkerToPath(marker *UpdateMarker, markerFile string, shouldFsync bool)
 	if err != nil {
 		return err
 	}
+	if isSafe {
```

Comment:
Same isSafe question as for the load. If we want to allow reading/writing the file concurrently with other updates, we can abstract the `Locker` and pass a different implementation that won't lock the lockfile